e2e/test/scenarios/onboarding/setup/setup.cy.spec.ts

```diff
@@ -406,44 +367,129 @@ describeWithSnowplow("scenarios > setup", () => {
+    // 4 - setup/step_seen  "language"
+    expectGoodSnowplowEvent({
+      event: "step_seen",
+      step_number: 1,
+      step: "language",
+    });
+    selectPreferredLanguageAndContinue();
 
+    // 5 - setup/step_seen "user_info"
+    expectGoodSnowplowEvent({
+      event: "step_seen",
+      step_number: 2,
+      step: "user_info",
+    });
     cy.findByTestId("setup-forms").within(() => {
-      // 4 - setup/step_seen  "language"
-      cy.findByText("What's your preferred language?");
-      cy.findByText("Next").click();
-
-      // 5 - setup/step_seen "user_info"
-      cy.findByText("What should we call you?");
-      cy.findByLabelText("Email").type(admin.email);
-      cy.findByLabelText("Company or team name").type("Epic Team");
-      cy.findByLabelText("Create a password").type(admin.password);
-      cy.findByLabelText("Confirm your password").type(admin.password);
-      cy.button("Next").click();
+      fillUserAndContinue({ firstName: null, lastName: null });
 
       cy.findByText("What will you use Metabase for?").should("exist");
       // 6 - setup/step_seen "usage_question"
-
+      expectGoodSnowplowEvent({
+        event: "step_seen",
+        step_number: 3,
+        step: "usage_question",
+      });
       cy.button("Next").click();
 
-      // 7 setup/usage_reason_selected
-      // 8 setup/step_seen "database"
+      // 7 - setup/usage_reason_selected
+      expectGoodSnowplowEvent({
+        event: "usage_reason_selected",
+        usage_reason: "self-service-analytics",
+      });
+      // 8 - setup/step_seen "db_connection"
+      expectGoodSnowplowEvent({
+        event: "step_seen",
+        step_number: 4,
+        step: "db_connection",
+      });
+      cy.findByText("I'll add my data later").click();
+      // 9 - setup/add_data_later_clicked
+      expectGoodSnowplowEvent({
+        event: "add_data_later_clicked",
+      });
+      // 10 - setup/step_seen "data_usage"
+      expectGoodSnowplowEvent({
+        event: "step_seen",
+        step_number: 5,
+        step: "data_usage",
+      });
+
+      cy.findByRole("button", { name: "Finish" }).click();
+      // 11 - new_user_created (from BE)
+
+      // 12- setup/step_seen "completed"
+      expectGoodSnowplowEvent({
+        event: "step_seen",
+        step_number: 6,
+        step: "completed",
+      });
     });
 
-    expectGoodSnowplowEvents(8);
+    expectGoodSnowplowEvents(12);
   });
 
   it("should ignore snowplow failures and work as normal", () => {
     blockSnowplow();
     cy.visit(`/setup`);
-    cy.findByTestId("welcome-page").within(() => {
-      cy.findByText("Welcome to Metabase");
-      cy.findByTextEnsureVisible("Let's get started").click();
-    });
+    skipWelcomePage();
 
     expectGoodSnowplowEvents(1);
   });
 });
+
+const skipWelcomePage = () => {
+  cy.findByTestId("welcome-page").within(() => {
+    cy.findByText("Welcome to Metabase");
+    cy.findByTextEnsureVisible("Let's get started").click();
```

Comment:
If I remember correctly, we couldn't click invisible elements in Cypress anyway. Remember when sometimes you need to pass `force: true` when the element isn't visible on the screen e.g. when it's outside viewport, has 0x0 in size, or actually not be visible to user's eyes.

This is redundant in my opinion. `cy.findByText("blabla").click()` should do the same thing. Although it's nice to be explicit about the assertions, so I'm not suggesting you change anything. It's just that personally, I wouldn't assert an element visibility before clicking it.